p2p/subscriber_metrics.go

```diff
@@ -0,0 +1,90 @@
+package p2p
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type subscriberMetrics struct {
+	headerPropagationTime metric.Float64Histogram
+	headerReceivedNum     metric.Int64Counter
+	subscriptionNum       metric.Int64UpDownCounter
+}
+
+func newSubscriberMetrics() *subscriberMetrics {
+	headerPropagationTime, err := meter.Float64Histogram(
+		"header_p2p_subscriber_propagation_time",
+		metric.WithDescription("header propagation time"),
+	)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
I made this on purpose. I looked up and down the stack and the only error types of errors that might be returned are those related to us choosing the right aggregation type for the right instruments and various naming conflicts. All these are programmer/code errors, so I decided to panic here.

Reply:
It doesn't allow application to stop gracefully. Lets just propagate it up the stack so it can shut down other components properly

Reply:
Actually, you have a point. I treated that as it is `init`, while it's not. This would run after other components are started, so gracefully shutdown is a must